hedera-node/docs/design/services/token-service/airdrops/token-claim-airdrop.md

```diff
@@ -0,0 +1,160 @@
+        - Confirm that for the given pending airdrops ids in the transaction there are corresponding pending transfers existing in state
+        - Check if the sender has sufficient balance to fulfill the airdrop
+    - Handle:
+        - Any additional validation depending on config or state i.e. semantics checks
+        - The business logic for claiming pending airdrops
+            - We need to create a token association between each `receiver_id` and  `token_reference`, future rents for token association slot should be paid by `receiver_id`
+            - Then we should transfer the claimed tokens to each `receiver_id`
+                - We can dispatch synthetic crypto transfer for this, but we must skip the assessment of custom fees
+        - Token transfers and associations should be externalized using the `tokenTransferLists` and `automatic_token_associations` fields in the transaction record
+    - Fees calculation
+
+### Zero-Balance accounts
+
+An account with no open auto-association slots can receive airdrops but must send a `TokenClaimAirdrop` transaction, which requires a payer. If the account has zero hbars, then it can still claim the transfer if someone else is willing to pay for that transaction. For example, a Dapp could be the payer on the transaction. Both the Dapp and the account must sign the transaction.
+
+### Hollow accounts
+
+Any existing hollow accounts that were created before [HIP-904](https://hips.hedera.com/hip/hip-904) will have no or limited number of `maxAutoAssociations` depending on if they were created with HBAR or token transfer respectively. That means an airdrop of unassociated tokens to such accounts will result in a pending transfer.
+Performing `TokenClaimAirdrop` for such hollow account will also complete the account by setting its key which will obtained from the required transaction signature. Completing the hollow account should not modify the `maxAutoAssociations` on the account. That should always be an explicit step by a user.
+
+## Acceptance Tests
+
+All of the expected behaviour described below should be present only if the new `TokenClaimAirdrop` feature flag is enabled.
+
+- Given existing pending airdrop in state when valid `TokenClaimAirdrop` transaction containing entry for the same pending airdrop is performed then the `TokenClaimAirdrop` should succeed resulting in:
+    - the tokens being claimed should be automatically associated with the `receiver_id` account
+    - the tokens being claimed should be transferred to the `receiver_id` account
+    - the pending airdrop should be removed from state
+- Given a successful `TokenClaimAirdrop`  transaction having a hollow account as `receiver_id` should also complete the account without modifying its `maxAutoAssociations` value
+- Given successful  `TokenClaimAirdrop` when `TokenClaimAirdrop` for the same airdrop is performed then the `TokenClaimAirdrop` should fail
```

Comment:
We can explicitly say that the `second` `TokenClaimAirdrop` should fail.

Reply:
Updated.